blaze-client/src/main/scala/org/http4s/client/blaze/ConnectionBuilder.scala

```diff
@@ -0,0 +1,16 @@
+package org.http4s.client.blaze
+
+import scala.concurrent.Future
+
+import org.http4s.Request
+
+import scalaz.concurrent.Task
+
+trait ConnectionBuilder {
+
+  /** Free resources associated with this client factory */
+  def shutdown(): Task[Unit]
+
+  /** Attempt to make a new client connection */
+  def makeClient(req: Request): Future[BlazeClientStage]
```

Comment:
This API is exposing both a Task and a Future.  Task is more in line with the rest of the API.